tools/config/__init__.py

```diff
@@ -610,24 +610,71 @@ def _get_mem_specs(self, memories, cmsis_part, exception_text):
                 continue
         raise ConfigException(exception_text)
 
-    @property
-    def rom(self):
-        """Get rom information as a pair of start_addr, size"""
+    def _get_all_rom_regions(self, all_regions=False):
+        """Get information of all available rom regions as a pair of [Region, [start_addr, size]]"""
         # Override rom_start/rom_size
         #
         # This is usually done for a target which:
         # 1. Doesn't support CMSIS pack, or
         # 2. Supports TrustZone and user needs to change its flash partition
+        rom_regions = {}
+        # Counter to keep track of ROM memories supported by target
+        rom_active_memories = 0
+        rom_memories = ['IROM1', 'PROGRAM_FLASH', 'IROM2']
         cmsis_part = self._get_cmsis_part()
-        rom_start, rom_size = self._get_mem_specs(
-            ["IROM1", "PROGRAM_FLASH"],
-            cmsis_part,
-            "Not enough information in CMSIS packs to build a bootloader "
-            "project"
-        )
-        rom_start = int(getattr(self.target, "mbed_rom_start", False) or rom_start, 0)
-        rom_size = int(getattr(self.target, "mbed_rom_size", False) or rom_size, 0)
-        return (rom_start, rom_size)
+        for memory in rom_memories:
+            try:
+                rom_start, rom_size = self._get_mem_specs(
+                memory.split(),
+                cmsis_part,
+                "Not enough information in CMSIS packs to build a bootloader "
+                "project"
+                )
+                rom_start = int(rom_start, 0)
+                rom_size = int(rom_size, 0)
+                if all_regions == False:
+                    return rom_start, rom_size
+                if memory=='IROM1' or memory=='PROGRAM_FLASH':
+                    memory = 'ROM'
+                else:
+                    rom_active_memories += 1
+                    memory = 'ROM'+str(rom_active_memories)
```

Comment:
Please either use string formatting:
```suggestion
                    memory = 'ROM{}'.format(str(rom_active_memories))
```

or add a space around the `+`.